src/algorithmForNN/pointCloudUtils.py

```diff
@@ -0,0 +1,296 @@
+
+    current_list = []
+    for i in range(4):
+        current_list.append(
+            sqrt(pow(rval[1][0] - rval[i][0], 2) + pow(rval[1][1] - rval[i][1], 2))
+        )
+
+    current_list.sort()
+
+    return current_list[1] * current_list[2]
+
+
+def segment_all_planes_from_point_cloud(
+    point_cloud: o3d.geometry.PointCloud,
+    min_count_of_points: int,
+    min_area_of_plane: float,
+    distance: float = 0.1,
+) -> list:
+    """
+    Function allows you to get all planes from a given point cloud
+    param: distance - Max distance a point can be from the plane model, and still be considered an inlier.
+    """
+    all_planes = []
+    inlier_cloud, outlier_cloud, plane_model = segment_plane_from_point_cloud(
+        point_cloud, distance
+    )
+    while outlier_cloud.has_points():
+        if (
+            len(inlier_cloud.points) > min_count_of_points
+            and get_area_of_plane(inlier_cloud, plane_model) > min_area_of_plane
+        ):
+            all_planes.append(inlier_cloud)
+        inlier_cloud, outlier_cloud, plane_model = segment_plane_from_point_cloud(
+            outlier_cloud, distance
+        )
+    if (
+        inlier_cloud.has_points()
+        and len(inlier_cloud.points) > min_count_of_points
+        and get_area_of_plane(inlier_cloud, plane_model) > min_area_of_plane
+    ):
+        all_planes.append(inlier_cloud)
+
+    return all_planes
+
+
+def create_dictionary_of_point_cloud(point_cloud: o3d.geometry.PointCloud) -> dict:
+    """
+    Function returns a dictionary of points, with which you can get indices faster
+    """
+    result_dictionary = {}
+    numpy_main_point_cloud = convert_point_cloud_to_numpy_array(point_cloud)
+
+    for idx, point in enumerate(numpy_main_point_cloud):
+        triple = (point[0], point[1], point[2])
+        result_dictionary[triple] = idx
+
+    return result_dictionary
+
+
+def append_indexes_list_of_points(
+    main_index_list: list, current_dict: dict, temp_point_cloud: np.ndarray
+) -> list:
+    """
+    Function finds the indices of the required points
+    """
+    for point in temp_point_cloud:
+        triple = (point[0], point[1], point[2])
+        main_index_list.append(current_dict[triple])
+    return main_index_list
+
+
+def create_label_file(
+    current_snapshot: str,
+    path_to_label_file: str,
+    label_list: list,
+    path_to_new_label_file: str,
+    min_count_of_points: int,
+    min_area_of_plane: float,
+):
+    """
+    Function creates a file with binary labels
+    """
+    main_point_cloud = get_point_cloud_from_bin_file(current_snapshot)
+    point_cloud_size = len(main_point_cloud.points)
+    current_dict = create_dictionary_of_point_cloud(main_point_cloud)
+    current_point_cloud = create_point_cloud_by_label_list(
+        main_point_cloud, path_to_label_file, label_list
+    )
+    planes_list = segment_all_planes_from_point_cloud(
+        current_point_cloud, min_count_of_points, min_area_of_plane, 0.3
```

Comment:
0.3 looks too much. Have you read about LiDAR parameters of noise that is used in KITTI dataset?